ui/app/manager/src/pages/page-configuration.ts

```diff
@@ -314,28 +463,29 @@ export class PageConfiguration extends Page<AppStateKeyed> {
         };
     }
 
-    protected async getMapConfig(): Promise<{[id: string]: any}> {
+    protected async getMapConfig(): Promise<MapConfig> {
         const response = await manager.rest.api.MapResource.getSettings();
-        return (response.data.options as {[id: string]: any});
+        const { options, sources } = response.data as MapConfig;
```

Comment:
As mentioned it is possible for `MapResource.getSettings()` to return 204 no content, `response.data` then returns `""` empty string so a thruthy check could help as currently this method returns an invalid MapConfig like:
```
{
  "options": undefined,
  "sources": undefined
```

If no map config is returned then the configuration page cannot configure the map so disable this part of the UI